uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Upgrader/Step/1034.php

```diff
@@ -7,20 +7,54 @@
 
   /**
    * Adds the category field to the Absence Type table
+   * and setup option value for absence type categories
    *
    * @return bool
    */
   public function upgrade_1034() {
+    $this->up1034_createCategoryColumn();
+    $this->up1034_createCategoryOptionValues();
+
+    return TRUE;
+  }
+
+  /**
+   * Creates the category field in Absence Type table if not existing
+   */
+  private function up1034_createCategoryColumn() {
     $absenceTypeTable = AbsenceType::getTableName();
 
     if (!SchemaHandler::checkIfFieldExists($absenceTypeTable, 'category')) {
       CRM_Core_DAO::executeQuery("
         ALTER TABLE {$absenceTypeTable}
-        ADD category varchar(10) NOT NULL 
-        COMMENT 'This is used for grouping leave types.'
+        ADD category int unsigned NOT NULL COMMENT 'This is used for grouping leave types.'
       ");
     }
+  }
 
-    return TRUE;
+  /**
+   * Ensures absence type category option group and its values exist, creating one if not
+   */
+  private function up1034_createCategoryOptionValues() {
```

Comment:
The option group install file can be used here as well to install for existing sites.
See here: https://github.com/compucorp/civihr/blob/667735780c972d7e2a91d71eb570c91e03567aca/uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Upgrader/Step/1009.php#L22